Sources/Apollo/WebSocketTransport.swift

```diff
@@ -0,0 +1,342 @@
+import Foundation
+import Starscream
+
+fileprivate var sequenceNumber : Int = 0
+
+/// A network transport that uses web sockets requests to send GraphQL subscription operations to a server, and that uses the Starscream implementation of web sockets.
+public class WebSocketTransport: NetworkTransport, WebSocketDelegate {
+
+    final let PROTOCOLS = ["graphql-ws"]
+    
+    var websocket: WebSocket? = nil
+    var error : Error? = nil
+    
+    let serializationFormat = JSONSerializationFormat.self
+    
+    var reconnect : Bool = false
+    var acked : Bool = false
+    var reconnected : Bool = false
+    
+    private var queue: [String] = []
+    private var params: [String:String]?
+    
+    private var subscribers = [String: (JSONObject?, Error?) -> Void]()
+    private var subscriptions : [String: String] = [:]
+    
+    private let sendOperationIdentifiers: Bool
```

Comment:
Some formatting inconsistencies above with spaces before/after colons. Perhaps it's time to include SwiftLint.